### PR TITLE
Added Half-Normal & Log-Normal Distributions

### PR DESCRIPTION
**Pull request by mschrader15**
I had a use case for a mixture model of half-normal & lognormal distributions. I did a quick implementation by sub-classing the normal distribution.

Missing tests, but this suffices for my use-case

### CODE REVIEW

**Comment by jmschrei** [comment]
Nice, thanks. I'll look into merging this soon.

File: pomegranate/distributions/lognormal.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	The parameters are the mu and sigma of the normal distribution, which 
+	is the the exponential of the log normal distribution. This
+	distribution can assume that features are independent of the others if
+	the covariance type is 'diag' or 'sphere', but if the type is 'full' then
+	the features are not independent.
+
+	There are two ways to initialize this object. The first is to pass in
+	the tensor of probablity parameters, at which point they can immediately be
+	used. The second is to not pass in the rate parameters and then call
+	either `fit` or `summary` + `from_summaries`, at which point the probability
```
Comment (jkleckner) [inline pomegranate/distributions/lognormal.py:34]: Nit: sumary -> summarize

File: pomegranate/distributions/halfnormal.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    distribution is sqrt(2/pi) * sigma, and the variance is (1 - 2/pi) * sigma^2.
+
+    This distribution can assume that features are independent of the others if
+    the covariance type is 'diag' or 'sphere', but if the type is 'full' then
+    the features are not independent.
+
+    There are two ways to initialize this object. The first is to pass in
+    the tensor of probablity parameters, at which point they can immediately be
+    used. The second is to not pass in the rate parameters and then call
+    either `fit` or `summary` + `from_summaries`, at which point the probability
```
Comment (jkleckner) [inline pomegranate/distributions/halfnormal.py:39]: Ditto

**Comment by jmschrei** [comment]
Sorry for the delay in approval. Would you mind adding in unit tests, of the style of the other distributions?

**Comment by jmschrei** [comment]
Otherwise this LGTM

**Pull request merged**